save-plugins/warn-plugin/src/commonMain/kotlin/com/saveourtool/save/plugin/warn/WarnPlugin.kt

```diff
@@ -258,30 +258,52 @@ class WarnPlugin(
         )
     }.asSequence()
 
-    @Suppress("TooGenericExceptionCaught", "SwallowedException")
+    @Suppress(
+        "TooGenericExceptionCaught",
+        "SwallowedException",
+        "TOO_LONG_FUNCTION"
+    )
     private fun collectExpectedWarnings(
         generalConfig: GeneralConfig,
         warnPluginConfig: WarnPluginConfig,
         originalPaths: List<Path>,
         copyPaths: List<Path>,
         workingDirectory: Path,
-    ): WarningMap = if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
-        val warningsFromSarif = try {
-            collectWarningsFromSarif(warnPluginConfig, originalPaths, fs, workingDirectory)
-        } catch (e: Exception) {
-            throw SarifParsingException("We failed to parse sarif. Check the your tool generation of sarif report, cause: ${e.message}", e.cause)
-        }
-        copyPaths.associate { copyPath ->
-            copyPath.name to warningsFromSarif.filter { it.fileName == copyPath.name }
+    ): WarningMap {
+        val expectedWarningsFileName: String by lazy {
+            warnPluginConfig.expectedWarningsFileName
+                ?: throw IllegalArgumentException("<expectedWarningsFileName> is not provided for expectedWarningsFormat=${warnPluginConfig.expectedWarningsFormat}")
```

Comment:
now we always expect the file? may be restore to the
`if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF || ExpectedWarningsFormat.Plain `
?

Reply:
it's a lazy variable to re-use this variable for `ExpectedWarningsFormat.SARIF` and `ExpectedWarningsFormat.PLAIN`

Reply:
oh, I see